### PR TITLE
Fixed and changed something

### PR DESCRIPTION
**Pull request by fubaWoW**
First at all -> very nice AddOn ;)  
Hopefully i've fixed all the Errors!

* Fixed Weapon Swap when entering Combat in Fishing
* Fixed wrong Weapon/Offand Equiped after stop Fishing
* MrFish will now be disabled when entering Combat  
  You need enable it manually after Comabt again
* Changed to use ItemLink instead of ItemID to Store and Restore Weapons because of Enchants and Sockets  
  This will prevent to equip the Wrong Item with the same Name but different Itemlevel, Enchant or Socket(s)
* Added a Delay of 0.5 Seconds before the Fishing Button appears  
  So the Bonus will now shown correctly after equip a Fishing Pole

### CODE REVIEW

File: MrFish.lua [hunk 1]
```diff
@@ -266,13 +267,13 @@ function addon:StartFishFrame(atCursor)
 	start.Amount:SetFormattedText(TRADESKILL_RANK_WITH_MODIFIER,fishingSkill,fishingBonus,fishingCap)
 end
 function addon:StopFishFrame(show)
-	local body,main,off='/stopcasting',weapons[INVSLOT_MAINHAND].name,weapons[INVSLOT_OFFHAND].name
+	local body,main,off='/stopcasting',weapons[INVSLOT_MAINHAND].link,weapons[INVSLOT_OFFHAND].link
 	if (main and off) then
-			body=format("/stopcasting\n/equip %s\n/equip %s",main,off)
+		body=format("/stopcasting\n/equipslot 16 %s\n/equipslot 17 %s",main,off)
```
Comment (alar77) [inline MrFish.lua]: Please, do not use hardcoded values for inventory/equipment slot

File: MrFish.lua [hunk 2]
```diff
@@ -266,13 +267,13 @@ function addon:StartFishFrame(atCursor)
...
 	elseif (main or off) then
-			body=format("/stopcasting\n/equip %s",main or off)
+		body=format("/stopcasting\n/equipslot 16 %s",main or off)
```
Comment (alar77) [inline MrFish.lua]: Please, do not use hardcoded values for inventory/equipment slot

File: MrFish.lua [hunk 3]
```diff
@@ -149,39 +149,40 @@ function addon:GetFishingPole()
 		end
 	end
 	if (maxname) then
-		self:Print(format(L["Found %s"],maxname))
+		self:Print(format(L["Best Fishing Pole: %s"],maxname))
```
Comment (alar77) [inline MrFish.lua]: In order to change a dictinaly value, plese customize localization instead of changing the keyword

**Review by alar77 — CHANGES_REQUESTED** [review]
Many thanks for the PR, could you check my comments?

File: MrFish.lua (hunk 1 above)
Comment (fubaWoW) [reply MrFish.lua]: It **must** be set to the right slot becasue there **can** be two similar Items (Weapons) with the Same Name but different Enchants, Sockets, Quality or Itemlevel and they **must** be placed at the correct slot when restoring Weapons!  
  
The `/equip` command does equip Items randomly.

File: MrFish.lua (hunk 2 above)
Comment (fubaWoW) [reply MrFish.lua]: see comment above

File: MrFish.lua (hunk 3 above)
Comment (fubaWoW) [reply MrFish.lua]: There is no localisation in my localization.lua.
Maybe it will be handled at Curseforge.
I'll restore the Original Value.

**Comment by fubaWoW** [comment]
Restored Localisation Value

**Comment by fubaWoW** [comment]
Well, everything should be done now.
Good you did not release it because i forgot to add the second delay for "StartFishFrame" :D

**Review by alar77 — APPROVED** [review]

**Pull request merged**